spec/isocalendar.html

```diff
@@ -151,112 +151,153 @@ <h1>Temporal.ISO8601Calendar.prototype.dateDifference ( _one_, _two_, _options_
       </emu-alg>
     </emu-clause>
 
     <emu-clause id="sec-temporal.iso8601calendar.prototype.dayofweek">
-      <h1>Temporal.ISO8601Calendar.prototype.dayOfWeek ( _date_ )</h1>
+      <h1>Temporal.ISO8601Calendar.prototype.dayOfWeek ( _dateOrDateTime_ )</h1>
       <p>
-        The `dayOfWeek` method takes one argument _date_.
+        The `dayOfWeek` method takes one argument _dateOrDateTime_.
         The following steps are taken:
       </p>
       <emu-alg>
-        1. <mark>TODO.</mark>
+        1. Perform ? RequireInternalSlot(_dateOrDateTime_, [[ISOYear]]).
+        1. Perform ? RequireInternalSlot(_dateOrDateTime_, [[ISOMonth]]).
+        1. Perform ? RequireInternalSlot(_dateOrDateTime_, [[ISODay]]).
+        1. Return ! ToDayOfWeek(_dateOrDateTime_.[[ISOYear]], _dateOrDateTime_.[[ISOMonth]], _dateOrDateTime_.[[ISODay]]).
       </emu-alg>
     </emu-clause>
 
     <emu-clause id="sec-temporal.iso8601calendar.prototype.dayofyear">
-      <h1>Temporal.ISO8601Calendar.prototype.dayOfYear ( _date_ )</h1>
+      <h1>Temporal.ISO8601Calendar.prototype.dayOfYear ( _dateOrDateTime_ )</h1>
       <p>
-        The `dayOfYear` method takes one argument _date_.
+        The `dayOfYear` method takes one argument _dateOrDateTime_.
         The following steps are taken:
       </p>
       <emu-alg>
-        1. <mark>TODO.</mark>
+        1. Perform ? RequireInternalSlot(_dateOrDateTime_, [[ISOYear]]).
+        1. Perform ? RequireInternalSlot(_dateOrDateTime_, [[ISOMonth]]).
+        1. Perform ? RequireInternalSlot(_dateOrDateTime_, [[ISODay]]).
+        1. Return ! ToDayOfYear(_dateOrDateTime_.[[ISOYear]], _dateOrDateTime_.[[ISOMonth]], _dateOrDateTime_.[[ISODay]]).
       </emu-alg>
     </emu-clause>
 
     <emu-clause id="sec-temporal.iso8601calendar.prototype.weekofyear">
-      <h1>Temporal.ISO8601Calendar.prototype.weekOfYear ( _date_ )</h1>
+      <h1>Temporal.ISO8601Calendar.prototype.weekOfYear ( _dateOrDateTime_ )</h1>
       <p>
-        The `weekOfYear` method takes one argument _date_.
+        The `weekOfYear` method takes one argument _dateOrDateTime_.
         The following steps are taken:
       </p>
       <emu-alg>
-        1. <mark>TODO.</mark>
+        1. Perform ? RequireInternalSlot(_dateOrDateTime_, [[ISOYear]]).
+        1. Perform ? RequireInternalSlot(_dateOrDateTime_, [[ISOMonth]]).
+        1. Perform ? RequireInternalSlot(_dateOrDateTime_, [[ISODay]]).
+        1. Return ! ToWeekOfYear(_dateOrDateTime_.[[ISOYear]], _dateOrDateTime_.[[ISOMonth]], _dateOrDateTime_.[[ISODay]]).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal.iso8601calendar.prototype.daysinweek">
+      <h1>Temporal.ISO8601Calendar.prototype.daysInWeek ( _dateOrDateTime_ )</h1>
+      <p>
+        The `daysInWeek` method takes one argument _dateOrDateTime_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Perform ? RequireInternalSlot(_dateOrDateTime_, [[ISOYear]]).
```

Comment:
Should it require ISOMonth and ISODay as well? Otherwise it wouldn't throw if given a YearMonth.